packages/libs/error-stack/src/report.rs

```diff
@@ -211,13 +262,112 @@ impl<C> Report<C> {
+    #[must_use]
+    #[cfg(all(nightly, feature = "std"))]
+    #[deprecated = "a report might contain multiple backtraces, use `request_ref::<Backtrace>()` \
+                    instead"]
+    pub fn backtrace(&self) -> Option<&Backtrace> {
+        self.request_ref::<Backtrace>().next()
+    }
+
+    #[allow(missing_docs)]
+    #[must_use]
+    #[cfg(feature = "spantrace")]
+    #[cfg_attr(
+        nightly,
+        deprecated = "a report might contain multiple spantraces, use \
+                      `request_ref::<SpanTrace>()` instead"
+    )]
+    #[cfg_attr(
+        not(nightly),
+        deprecated = "a report might contain multiple spantraces, use \
+                      `frames().filter(Frame::downcast_ref::<SpanTrace>)` instead"
+    )]
+    pub fn span_trace(&self) -> Option<&SpanTrace> {
+        #[cfg(nightly)]
+        return self.request_ref::<SpanTrace>().next();
+
+        #[cfg(not(nightly))]
+        return self.downcast_ref::<SpanTrace>();
+    }
+
+    /// Merge two [`Report`]s together
+    ///
+    /// This function appends the [`current()`] frames of the other [`Report`] to the [`current()`]
+    /// frames of this report.
+    /// Meaning `A.extend_one(B) -> A.current() = A.current() + B.current()`
+    ///
+    /// [`current()`]: Self::current
+    ///
+    /// ```rust
+    /// use std::{
+    ///     fmt::{Display, Formatter},
+    ///     path::Path,
+    /// };
+    ///
+    /// use error_stack::{Context, Report};
+    /// # #[cfg(all(not(miri), feature = "std"))]
+    /// use error_stack::{IntoReport, ResultExt};
+    ///
+    /// #[derive(Debug)]
+    /// struct IoError;
+    ///
+    /// impl Display for IoError {
+    ///     # fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+    ///     #     const _: &str = stringify!(
+    ///             ...
+    ///     #     );
+    ///     #     f.write_str("Io Error")
+    ///     # }
+    /// }
+    ///
+    /// # impl Context for IoError {}
+    ///
+    /// # #[allow(unused_variables)]
+    /// fn read_config(path: impl AsRef<Path>) -> Result<String, Report<IoError>> {
+    ///     # #[cfg(any(miri, not(feature = "std")))]
+    ///     # return Err(error_stack::report!(IoError).attach_printable("Not supported"));
+    ///     # #[cfg(all(not(miri), feature = "std"))]
+    ///     std::fs::read_to_string(path.as_ref())
+    ///         .into_report()
+    ///         .change_context(IoError)
+    /// }
+    ///
+    /// let mut error1 = read_config("config.txt").unwrap_err();
+    /// let error2 = read_config("config2.txt").unwrap_err();
+    /// let mut error3 = read_config("config3.txt").unwrap_err();
+    ///
+    /// error1.extend_one(error2);
+    /// error3.extend_one(error1);
+    ///
+    /// // ^ This is equivalent to:
+    /// // error3.extend_one(error1);
+    /// // error3.extend_one(error2);
+    /// ```
+    ///
+    /// [`extend_one()`]: Self::extend_one
+    pub fn extend_one<T>(&mut self, mut report: Report<T>) {
```

Comment:
Let's not allow extending from a `Report` with a different `Context`.
```suggestion
    pub fn extend_one(&mut self, mut report: Self) {
```